landoapi/transplant_client.py

```diff
@@ -0,0 +1,83 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import json
+import os
+import requests
+import requests_mock
+
+
+class TransplantClient:
+    """ A class to interface with Transplant's API. """
+
+    def __init__(self):
+        self.api_url = os.getenv('TRANSPLANT_URL') + '/api'
+
+    @requests_mock.mock()
+    def land(
+        self,
+        ldap_username,
+        tree,
+        patch,
+        destination,
+        push_bookmark,
+        pingback_url,
+        request,  # mocked requests
+        rev='',
+        commit_descriptions=''
+    ):
+        """ Sends a push request to Transplant API to land a revision.
+
+        Returns request_id received from Transplant API.
+        """
+        # Connect to stubbed Transplant service
+        request.post(
+            self.api_url + '/autoland',
+            text=json.dumps({
+                'request_id': 1
+            }),
+            status_code=202
+        )
+
+        result = self._POST(
+            '/autoland', {
+                'ldap_username': ldap_username,
+                'tree': tree,
+                'rev': rev,
+                'destination': destination,
+                'push_bookmark': push_bookmark,
+                'commit_descriptions': commit_descriptions,
+                'pingback_url': pingback_url
+            }
+        )
+
+        return result.get('request_id') if result else None
+
+    def _request(
+        self, url, data=None, params=None, method='GET', success_code=200
+    ):
+        data = data if data else {}
+        response = requests.request(
+            method=method,
+            url=self.api_url + url,
+            params=params,
+            data=data,
+            timeout=10
```

Comment:
Too early to know, but, we should ask around to make sure this timeout value isn't too short. I don't know how long transplant takes to do things :)